src/main/java/network/brightspots/rcv/ContestConfig.java

```diff
@@ -948,7 +941,7 @@ boolean isContinueUntilTwoCandidatesRemainEnabled() {
   }
 
   int getNumDeclaredCandidates() {
-    int size = getCandidateCodeList().size();
+    int size = candidateNames.size();
```

Comment:
How do we know this will be defined yet? Shouldn't we do `getCandidateNames()` here instead?

Reply:
This may be a java convention I'm unfamiliar with -- `getCandidateNames` doesn't do a null check, why would it help?

I can make this safer by:
1. `candidateNames == null ? 0 : candidateNames.size()`
2. Initializing it when declaring it
3. Initialize `candidateNames` in `getCandidateNames()` if needed.

Though, I'll note that the non-null assumption existed in the initial code (unless there's some magic Java does with getters that I'm unfamiliar with).

Reply:
I think it's just generally good practice to use the getter instead of accessing the variable directly, in case we ever want to change the behavior in the future.

I like the idea of making the getter check if it's not defined yet, and returning an empty set if it's not.